api.py

```diff
@@ -0,0 +1,59 @@
+import requests
+import json
+import urllib.parse
+
+class ApiController(object):
+    def __init__(self, url: str):
+        self.url = url if url[-1] != '/' else url[:-1]
+        self.endpoints = {
+            'levels': '/levels',
+            'groups': '/groups',
+            'schedules': '/schedules'
+        }
+
+    def error_check(self, endpoint, params = ''):
+        response = requests.get(self.url + self.endpoints[endpoint] + params).json()
+        status = 'success' if 'response' in response else 'error'
+        response = response['response'] if status == 'success' else response['error']
+        return {'status': status, 'response': response}
+
+    def get_levels(self):
+        return api.error_check('levels')
```

Comment:
вызов объекта внутри него..
у тебя так в нескольких местах

```suggestion
        return self.error_check('levels')
```